lightning/src/events/bump_transaction/mod.rs

```diff
@@ -436,7 +438,13 @@ where
 		&self, utxos: &[Utxo], claim_id: ClaimId, force_conflicting_utxo_spend: bool,
 		tolerate_high_network_feerates: bool, target_feerate_sat_per_1000_weight: u32,
 		preexisting_tx_weight: u64, input_amount_sat: Amount, target_amount_sat: Amount,
+		max_tx_weight: u64,
 	) -> Result<CoinSelection, ()> {
+		let change_script = self.source.get_change_script().await?;
+		let max_utxo_selection_weight = max_tx_weight
+			- preexisting_tx_weight
+			- (change_script.len() as u64 + 8 + 1) * WITNESS_SCALE_FACTOR as u64;
```

Comment:
I understand we're being conservative here by considering the change script weight even though we may not end up having one. Unfortunately, calling this too early before we know if we need it could lead to address inflation.